### ISSUE TITLE
[feature] Exclude directory except for certain file(s) in .nuclei-ignore

### ISSUE THREAD

**Issue by nirvana-msu** [projectdiscovery/nuclei#622]
**Is your feature request related to a problem? Please describe.**
It would be great if `.nuclei-ignore` allowed to ignore a directory with the exception of certain file(s).

**Describe the solution you'd like**
`.gitignore` provides this "except" functionality using [`!` prefix](https://stackoverflow.com/questions/987142/make-gitignore-ignore-everything-except-a-few-files) - ideally `.nuclei-ignore` should honour this synax as well.

You could even support the entire syntax of `.gitignore` (why re-invent the wheel) - there seems to be [plenty](https://www.google.com/search?q=gitignore+parser+go) of ready-to-use parsers for go.

**Comment by nirvana-msu** [comment]
Somewhat unrelated:  I tried placing `.nuclei-ignore` in my working directory (which should work, according to documentation) but somehow it didn't seem to get picked up.... perhaps I did something wrong, not sure. I then tried specifing `templates` / `exclude` using config file instead, and that also failed - see #623

**Comment by ehsandeep** [comment]
This should be a fixed release of [v3.2.2](https://github.com/projectdiscovery/nuclei/releases/tag/v2.3.2) along with nuclei config file, now config file is more preferable choice to add exclude templates as it supports multiple inputs.

```yaml
exclude:
  - dns/
  - headless/
  - network/mongodb-detect.yaml
```

Thanks again for reporting this.
